Server/htdocs/AppController/commands_RSM/utilities/RSMverifyBody.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+//***************************************************************************************
+// Structure
+//***************************************************************************************
+
+function checkBodyIsJsonObject($body)
+{
+  global $RSallowDebug;
+  if (!is_object($body)) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body must be a JSON object '{}'");
+    else returnJsonMessage(400, "");
+  }
+}
+
+function checkIsJsonObjectInArray($body)
+{
+  global $RSallowDebug;
+  if (!is_object($body)) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body array elements must be JSON objects '{}'");
+    else returnJsonMessage(400, "");
+  }
+}
+
+function checkIDsIsArray($body)
+{
+  global $RSallowDebug;
+  if (isset($body->IDs)) {
+    if (!is_array($body->IDs)) {
+      if ($RSallowDebug) returnJsonMessage(400, "Request body 'IDs' field must be an array");
+      else returnJsonMessage(400, "");
+    }
+  }
+}
+
+function checkPropertyIDsIsArray($body)
+{
+  global $RSallowDebug;
+  if (isset($body->propertyIDs)) {
+    if (!is_array($body->propertyIDs)) {
+      if ($RSallowDebug) returnJsonMessage(400, "Request body 'propertyIDs' field must be an array");
+      else returnJsonMessage(400, "");
+    }
+  }
+}
+
+function checkBodyIsArray($body)
+{
+  global $RSallowDebug;
+  if (!is_array($body)) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body must be an array '[]'");
+    else returnJsonMessage(400, "");
+  }
+}
+
+function checkIDsIsAnArray($body)
+{
+  global $RSallowDebug;
+  if (!is_array($body->IDs)) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body 'IDs' field must be an array '[]'");
+    else returnJsonMessage(400, "");
+  }
+}
+
+//***************************************************************************************
+// Contains
+//***************************************************************************************
+
+function checkBodyContainsID($body)
+{
+  global $RSallowDebug;
+  if (!(isset($body->ID))) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body must contain ID");
+    else returnJsonMessage(400, "");
+  }
+}
+
+function checkBodyContainsItemTypeId($body)
+{
+  global $RSallowDebug;
+  if (!(isset($body->itemTypeID))) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body must contain itemTypeID");
+    else returnJsonMessage(400, "");
+  }
+}
+
+function checkBodyContainsPropertyID($body)
+{
+  global $RSallowDebug;
+  if (!(isset($body->propertyID))) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body must contain propertyID");
+    else returnJsonMessage(400, "");
+  }
+}
+
+function checkBodyContainsItemTypeIDorPropertyIDs($body)
+{
+  global $RSallowDebug;
+  if (!isset($body->itemTypeID) and !isset($body->propertyIDs)) {
+    if ($RSallowDebug) returnJsonMessage(400, "Request body must contain at least field 'itemTypeID' or field 'propertyIDs'");
+    else returnJsonMessage(400, "");
+  }
+}
```

Comment:
The name of the function mentions "or" but the function contacins an "and" validation with two negations, which is confusing. It would be better to use "isset + or + isset" instead of "!isset and !isset"